extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/controlBar.js

```diff
@@ -0,0 +1,506 @@
+    evt.preventDefault();
+    this.props.controller.toggleFullscreen();
+  },
+
+  handleLiveClick: function(evt) {
+    evt.stopPropagation();
+    evt.cancelBubble = true;
+    evt.preventDefault();
+    this.props.controller.onLiveClick();
+    this.props.controller.seek(this.props.duration);
+  },
+
+  handleVolumeIconClick: function(evt) {
+    if (this.isMobile){
+      this.props.controller.startHideControlBarTimer();
+      evt.stopPropagation(); // W3C
+      evt.cancelBubble = true; // IE
+      var canShowVolumeSliderBar = !this.props.skinConfig.controlBar
+        || !this.props.skinConfig.controlBar.volumeControl
+        || this.props.skinConfig.controlBar.volumeControl.sliderVisible !== false;
+      if (canShowVolumeSliderBar && !this.props.controller.state.volumeState.volumeSliderVisible){
+        this.props.controller.showVolumeSliderBar();
+      }
+      else {
+        this.props.controller.handleMuteClick();
+      }
+    }
+    else{
+      this.props.controller.handleMuteClick();
+    }
+  },
+
+  handlePlayClick: function() {
+    this.props.controller.togglePlayPause();
+  },
+
+  handleShareClick: function() {
+    this.props.controller.toggleShareScreen();
+  },
+
+  handleQualityClick: function() {
+    if(this.props.responsiveView == this.props.skinConfig.responsive.breakpoints.xs.id) {
+      this.props.controller.toggleScreen(CONSTANTS.SCREEN.VIDEO_QUALITY_SCREEN);
+    } else {
+      this.toggleQualityPopover();
+      this.closeCaptionPopover();
+    }
+  },
+
+  toggleQualityPopover: function() {
+    this.props.controller.toggleVideoQualityPopOver();
+  },
+
+  closeQualityPopover: function() {
+    if(this.props.controller.state.videoQualityOptions.showVideoQualityPopover == true) {
+      this.toggleQualityPopover();
+    }
+  },
+
+  toggleCaptionPopover: function() {
+    this.props.controller.toggleClosedCaptionPopOver();
+  },
+
+  closeCaptionPopover: function() {
+    if(this.props.controller.state.closedCaptionOptions.showClosedCaptionPopover == true) {
+      this.toggleCaptionPopover();
+    }
+  },
+
+  closePopovers: function() {
+    this.closeCaptionPopover();
+    this.closeQualityPopover();
+  },
+
+  handleVolumeClick: function(evt) {
+    evt.preventDefault();
+    var newVolume = parseFloat(evt.target.dataset.volume);
+    this.props.controller.setVolume(newVolume);
+  },
+
+  handleDiscoveryClick: function() {
+    this.props.controller.toggleDiscoveryScreen();
+  },
+
+  handleMoreOptionsClick: function() {
+    this.props.controller.toggleMoreOptionsScreen(this.moreOptionsItems);
+  },
+
+  handleClosedCaptionClick: function() {
+    if(this.props.responsiveView == this.props.skinConfig.responsive.breakpoints.xs.id) {
+      this.props.controller.toggleScreen(CONSTANTS.SCREEN.CLOSEDCAPTION_SCREEN);
+    } else {
+      this.toggleCaptionPopover();
+      this.closeQualityPopover();
+    }
+  },
+
+  //TODO(dustin) revisit this, doesn't feel like the "react" way to do this.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1135)